File: spring-todomvc-rest/src/main/java/org/example/todomvc/config/WebSecurityConfig.java
package org.example.todomvc.config;

import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;

@EnableWebSecurity
public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
	@Override
	protected void configure(HttpSecurity http) throws Exception {
//		http.authorizeRequests()
//			.anyRequest().permitAll();
			
		http.authorizeRequests()
			.antMatchers("/", "/index.html", "/api/**").permitAll()
			.antMatchers("/protected/**").authenticated()
//			.anyRequest().authenticated()
			.and()
			.formLogin().loginPage("/login").permitAll()
		;
	}
	
	
	@Override
	protected void configure(AuthenticationManagerBuilder auth) throws Exception {
//		super.configure(auth);
		auth.inMemoryAuthentication().withUser("user").password("pass").roles("USER");
//		auth.userDetailsService(userDetailsService());
	}

//	@Override
//	@Bean
//	protected UserDetailsService userDetailsService() {
//		InMemoryUserDetailsManager manager = new InMemoryUserDetailsManager();
//		manager.createUser(
//				User.withDefaultPasswordEncoder().username("user").password("pass").roles("USER").build());
//		return manager;
//	}

}

File: htmx/src/main/kotlin/com/example/spring/htmx/config/WebSecurityConfig.kt
package com.example.spring.htmx.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
import org.springframework.security.web.SecurityFilterChain


@Configuration
@EnableWebSecurity
class WebSecurityConfig {

  @Bean
  @Throws(Exception::class)
  fun securityFilterChain(http: HttpSecurity): SecurityFilterChain {
    http
      .authorizeHttpRequests { requests ->
        requests
          .requestMatchers("/**").permitAll()
          .anyRequest().authenticated()
      }
//      .formLogin { form ->
//        form
//          .loginPage("/login")
//          .permitAll()
//      }
      .logout { logout -> logout.permitAll() }
    return http.build()
  }

/*
  @Bean
  fun userDetailsService(): UserDetailsService {
    val user: UserDetails = User.withDefaultPasswordEncoder()
      .username("user")
      .password("password")
      .roles("USER")
      .build()
    return InMemoryUserDetailsManager(user)
  }
*/

}
